### PR TITLE
Function/get malshab

### PR DESCRIPTION
**Pull request by lustigy96**

### CODE REVIEW

File: GetMalshab/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import getConnection from '../shared/mongoose';
+
+const functionName = 'getMalshab';
+
+const getMalshab: AzureFunction = async (context: Context, req: HttpRequest, user: User): Promise<HttpTriggerResult> => {
+    try {
+        const identityNumber = req.params.id;
+        if (!user.role || !identityNumber) {
+            throw new ValidationFunctionError(functionName, `malshab id or user is not spesfied`);
+        } else {
```
Comment (yonatandt) [inline GetMalshab/index.ts:17]: redundant `else`

File: GetMalshab/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            await getConnection();
+            const malshab = await MalshabModel.find({ identityNumber: identityNumber }).exec();
```
Comment (yonatandt) [inline GetMalshab/index.ts:19]:
```suggestion
            const malshab = await MalshabModel.find({ identityNumber }).exec();
```

File: GetMalshab/utils.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+export const malshabFilter = (malshab: Malshab & mongoose.Document<any>, allowedKeys: string[]): Partial<Malshab> | undefined => {
+    const filtered = Object.keys(malshab.toJSON())
+        .filter((key) => allowedKeys.includes(key))
+        .reduce((obj, key) => {
+            // eslint-disable-next-line no-param-reassign
+            obj[key] = malshab[key];
+            return obj;
+        }, {});
+    if (Object.keys(filtered).length) return filtered;
+    return undefined;
```
Comment (yonatandt) [inline GetMalshab/utils.ts:15]:
```suggestion
    return (Object.keys(filtered).length) ? filtered : undefined;
```

File: GetMalshab/permissions.ts [hunk 4]
```diff
@@ -0,0 +1,256 @@
+const rolePermissions = {
```
Comment (yonatandt) [inline GetMalshab/permissions.ts:1]: Not that good of a name,
Maybe call it `malshabFieldsByRole`.

File: GetMalshab/utils.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import * as mongoose from 'mongoose';
+import Malshab from '../shared/malshab/malshab.interface';
+import rolePermissions from './permissions';
+import { Role } from '../shared/user/user.interface';
+
+export const malshabFilter = (malshab: Malshab & mongoose.Document<any>, allowedKeys: string[]): Partial<Malshab> | undefined => {
```
Comment (yonatandt) [inline GetMalshab/utils.ts:6]: Add documentation for this function.

File: GetMalshab/utils.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        .reduce((obj, key) => {
+            // eslint-disable-next-line no-param-reassign
+            obj[key] = malshab[key];
+            return obj;
+        }, {});
+    if (Object.keys(filtered).length) return filtered;
+    return undefined;
+};
+
+export const getPermissions = (role: Role): string[] => {
```
Comment (yonatandt) [inline GetMalshab/utils.ts:18]: Add documentation for this function.
Also, consider changing the name to: `getAllowedMalshabFields`, or something better.

**Review by yonatandt — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: GetMalshab/index.ts (hunk 2 above)
Comment (yonatandt) [reply GetMalshab/index.ts:19]: Also, consider using `findOne` instead.

File: GetMalshab/index.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+const functionName = 'getMalshab';
+
+const getMalshab: AzureFunction = async (_context: Context, req: HttpRequest, user: User): Promise<HttpTriggerResult> => {
+    try {
+        const identityNumber = req.params.id;
+        if (!user.role || !identityNumber) {
+            throw new ValidationFunctionError(functionName, `malshab id or user is not spesfied`);
+        }
+        await getConnection();
+        const malshab = await MalshabModel.find({ identityNumber }).exec();
```
Comment (yonatandt) [inline GetMalshab/index.ts:19]: Use `findOne` instead of `find`.

**Review by yonatandt — CHANGES_REQUESTED** [review]

**Review by yonatandt — APPROVED** [review]

**Review by carmeltsadok — APPROVED** [review]

**Review by yonatandt — APPROVED** [review]

**Pull request merged**